django/contrib/admin/static/admin/js/calendar.js

```diff
@@ -187,25 +193,39 @@ depends on core.js for utility functions like removeChildren or quickElement
 
             this.drawCurrent();
         },
-        drawPreviousMonth: function() {
-            if (this.currentMonth === 1) {
-                this.currentMonth = 12;
-                this.currentYear--;
+        drawPreviousMonth: function(num) {
+            const captionlink = document.getElementById('yearcaption' + num);
+            // when we are in monthlist box, we have captionlink.
+            if (captionlink !== null) {                                    
```

Comment:
I fixed it and also changed some other `!==` of my codes to `!=` and also there was several `!==` from before in DateTimeShortcuts.js and calendar.js  and i think is unnecessary and should change to `!=`. for example in `if (window.getComputedStyle(document.body).direction !== 'rtl') {` or in `((year % 4) === 0) && ((year % 100) !== 0 )`, why we should check if types was unequal return true? suppose this `'0' !== 0` will be true, this may not raise problem but is not logical.
Also unexpected error in **JavaScript tests (pull_request)** for Inequality should fix?